pkg/web/client_test.go

```diff
@@ -74,3 +74,13 @@ func TestVersionFromManifest(t *testing.T) {
 	require.Equal(t, openAPISchema, version.OpenAPISchema)
 	require.Equal(t, cogConfig, version.CogConfig)
 }
+
+func TestVersionURLErrorWithoutR8IMPrefix(t *testing.T) {
+	_, err := newVersionURL("docker.com/thing/thing")
+	require.Error(t, err)
+}
+
+func TestVersionURLErrorWithout3Components(t *testing.T) {
+	_, err := newVersionURL("r8.im/username")
```

Comment:
I'd expect a more likely real situation would be 
```suggestion
	_, err := newVersionURL("username/test")
```